manifests/istio-telemetry/prometheus/templates/configmap.yaml

```diff
@@ -25,268 +24,132 @@ data:
         action: keep
         regex: istio-telemetry;prometheus
 
-    # Scrape config for envoy stats
-    - job_name: 'envoy-stats'
-      metrics_path: /stats/prometheus
-      kubernetes_sd_configs:
-      - role: pod
-
-      relabel_configs:
-      - source_labels: [__meta_kubernetes_pod_container_port_name]
-        action: keep
-        regex: '.*-envoy-prom'
-      - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
-        action: replace
-        regex: ([^:]+)(?::\d+)?;(\d+)
-        replacement: $1:15090
-        target_label: __address__
-      - action: labeldrop
```

Comment:
this is one thing we are losing here -- the dropping of labels on `envoy-stats` collection. this essentially, as best I can tell, reverts the PR that did that for 1.5.

This will result in `istio_` metrics having duplications like `{"app"="foo", "destination_app"="foo"}` etc., plus that pod template hash label, right?  And I think also add in some of the k8s labels like `kubernetes_namespace` and `pod_name`, right?

That will increase storage requirements and the overhead of bytes transferred, as well as some potential user confusion.

@mandarjog thoughts here?

Reply:
The easy fix without context of other (possible) changes that we are making is just to drop these labels in the config again here. This will mean that our prom config is "better" than out-of-the-box configs though, which would be especially bad if we move forward with the addon changes.

I think its somewhat reasonable to say that if you are a user that cares about the metrics size a lot, you can configure it to drop metrics. The unfortunate thing here is the confusion (app vs destination_app) which impacts new users as well

Reply:
@nrjpoddar responding here to 

> I think migration would be an issue for existing customers if they want to preserve the label set they have with 1.5 or 1.4. But tbh most of the customers are probably tinkering with the Prometheus config and not just using what Istio "demo" gives them, so their old settings will be preserved as long as they do the merge. For our customers, we provide our own version of Prometheus with custom config so this doesn't impact us.

I would agree I don't think the "production user" is impacted here. People who care about the scalability concerns of additional labels will have their own configs. So to me this is less important.

> The bigger impact is probably on UX and confusion due to similarly named labels, may be we can add a doc with link to the scrape config to fix it?

This I would agree. I am not sure how much of an impact this is for users to decide what we should do though.

it would be easy to add the label dropping back in theory but the problem is we then would either drop application metric labels (not desirable) or have a custom scrape config for envoy (what this entire PR is trying to avoid), so its pretty challenging